File: app/src/main/java/de/bitb/astroskop/injection/modules/ControllerModule.java
package de.bitb.astroskop.injection.modules;

import dagger.Module;
import dagger.Provides;
import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.controller.CircumstanceController;
import de.bitb.astroskop.controller.ProfileController;
import de.bitb.astroskop.injection.ApplicationScope;

@Module
public class ControllerModule {

    @Provides
    @ApplicationScope
    public CircumstanceController provideCircumstanceController(AstroApplication astroApplication) {
        return new CircumstanceController(astroApplication);
    }
    @Provides
    @ApplicationScope
    public ProfileController provideProfileController(AstroApplication astroApplication) {
        return new ProfileController(astroApplication);
    }

}

File: app/src/main/java/de/bitb/astroskop/injection/modules/ControllerModule.kt
package de.bitb.astroskop.injection.modules

import dagger.Module
import dagger.Provides
import de.bitb.astroskop.BaseApplication
import de.bitb.astroskop.controller.CircumstanceController
import de.bitb.astroskop.controller.ProfileController
import javax.inject.Singleton

@Module
class ControllerModule {

    @Provides
    @Singleton
    fun provideCircumstanceController(baseApplication: BaseApplication): CircumstanceController {
        return CircumstanceController(baseApplication)
    }

    @Provides
    @Singleton
    fun provideProfileController(baseApplication: BaseApplication): ProfileController {
        return ProfileController(baseApplication)
    }

}
